collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -56,15 +64,45 @@
 
     private FormMapViewModel viewModel;
 
-    private MapFragment map;
+    @VisibleForTesting public MapFragment map;
```

Comment:
I think it would be nice to use a factory for injection here rather the `MapFragment` itself. Google talked about introducing `FragmentFactory` in the near future (https://www.notion.so/Fragments-Past-Present-and-Future-f21afdfc5e5346f9bba1c89707ff3358) so this feels in keeping with where the framework is going as well.

Reply:
I should perhaps have given a bit more detail here about what I was running into. There's already a static `MapProvider` which uses general settings to build a configurator which can itself build a `MapFragment`. I think that ideally the `MapProvider` wouldn't be static but that will require (some) rework of all the geo components as far as I can see. Is that what you have in mind? Do you think it's appropriate to do here/now?

Reply:
Ah actually it looks [`FragmentFactory`](https://developer.android.com/reference/androidx/fragment/app/FragmentFactory) already exists and can be set on the `FragmentManager`. I'm not too in to the design of it - you basically create a `FragmentFactory` implementation that handles all the Fragments in your app.

I had a quick play with using this approach but I think you're right in that because we have multiple of the way the `MapFragment` is designed (it's not a fragment) this approach wouldn't work. Happy to go with your approach right now I think. Alternatively if you're up for it you could wrap the call to `MapProvider.createMapFragment` in a simple Factory that you can swap out in the test.

Reply:
> it's not a fragment

I guess you mean that the concrete classes are `Fragment`s but the interface itself doesn't include anything `Fragment`-like? That is quite odd now that you mention it. And I guess there's no way to require that subclasses be `Fragment`s since there is no `Fragment` interface? Yikes. Not sure what to do about this. It seems like semantically we want something like `MapFragment extends Fragment`. Or maybe `MapFragment` should be more of a fragment container?

> you could wrap the call to MapProvider.createMapFragment in a simple Factory that you can swap out in the test.

That makes sense. Feels a little inception-like but will play with it!

Reply:
> It seems like semantically we want something like `MapFragment extends Fragment`. Or maybe `MapFragment` should be more of a fragment container?

Yeah my first though is that `MapFragment` should extend `Fragment` and should have a `setMap`/`getMap` that takes the interface we currently call `MapFragment` (The interface formerly known as...)

Reply:
> `MapFragment` should extend `Fragment`

But that's not possible because there's no `Fragment` interface, right? Or am I missing something...

Reply:
I mean we make a MapFragment extends the Android Fragment class and that it can support any of the mapping engines (maybe by using a Fragment container). That's future us's problem though - not for this PR.